From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 09:25:22 -0700
Subject: [PATCH] mm/damon/Kconfig: enable CONFIG_DAMON by default

DAMON has been useful in productions for years.  As of this writing, a
few major distros including Alma, Amazon, Android, CentOS, Debian,
Fedora, and Oracle are build-enabling (set CONFIG_DAMON y).  Enabling it
by default may help ther configuration setup easier.

Build-enabling DAMON does not introduce a real risk since it makes no
behavioral change by default.  It requires users to make explicit
requests to do something.  Only one potential risk is making the size of
the kernel little bit larger.  On a configuration, it increases
resulting kernel package size 337,685 bytes, which is about 0.1 % of the
final package file.

Hence, the benefit of enabling CONFIG_DAMON out-weighs the potential
risk.  Set CONFIG_DAMON by default.

[1] https://oracle.github.io/kconfigs/?config=UTS_RELEASE&config=DAMON

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index c93d0c56b963..551745df011b 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -4,6 +4,7 @@ menu "Data Access Monitoring"
 
 config DAMON
 	bool "DAMON: Data Access Monitoring Framework"
+	default y
 	help
 	  This builds a framework that allows kernel subsystems to monitor
 	  access frequency of each memory region. The information can be useful
-- 
2.39.5

